### PR TITLE
Added 5 Solutions

### PR DESCRIPTION
**Pull request by Yash1256**
@priyanshisharma  Added 5 Solutions to Issue #54 , April Coding Challenge and make it's README.md file

### CODE REVIEW

**Review by UtkarshChaurasia — APPROVED** [review]
All the solutions are working perfectly. @priyanshisharma you can merge this PR.

File: April/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# April-Leetcoding-Challenge
+### This repository contains the solutions I submitted during the June Leetcoding Challenge. Below are the links of questions on LeetCode if you wish to try them yourself!
```
Comment (priyanshisharma) [inline April/README.md:2]:
```suggestion
### This repository contains the solutions for the April Leetcoding Challenge. Below are the links of questions on LeetCode if you wish to try them yourself!
```

**Review by priyanshisharma — COMMENTED** [review]
@UtkarshChaurasia i also need to give the people who claimed the issue the first chance. And I also wished to restrict participation to 2 problems per person and mentioned and assigned in #54. I don't think I have assigned any questions to @Yash1256 there.

Diameter of binary tree has been taken by @Aastha26-lab , group anagrams and move zeroes by @shibrahmachari , maximum subarray by @RajawatBanna , number of islands by @Deepak290701 

If none of these make their PRs I would be happy to merge these.

**Comment by Yash1256** [comment]
Thanks @priyanshisharma for understanding.

**Comment by Yash1256** [comment]
Sorry my bad, By mistake that branch got deleted, extremely sorry for that

**Comment by UtkarshChaurasia** [comment]
> @UtkarshChaurasia i also need to give the people who claimed the issue the first chance. And I also wished to restrict participation to 2 problems per person and mentioned and assigned in #54. I don't think I have assigned any questions to @Yash1256 there.
> 
> Diameter of binary tree has been taken by @Aastha26-lab , group anagrams and move zeroes by @shibrahmachari , maximum subarray by @RajawatBanna , number of islands by @Deepak290701
> 
> If none of these make their PRs I would be happy to merge these.

Sorry @priyanshisharma I did not know this. I was just trying to help by reviewing as I have already solved these questions.

**Comment by priyanshisharma** [comment]
Not a problem @UtkarshChaurasia , though I did expect you to know that due to your comment on #6 .
Not a problem @Yash1256 I'll reassign the not solved ones on 7th October.

**Comment by priyanshisharma** [comment]
Since @RajawatBanna hasn't made a PR for maximum subarray, you may delete rest of the solutions and I'll be happy to merge your maximum subarray solution.

**Comment by priyanshisharma** [comment]
@Yash1256 Feel free to comment if there are any doubts.

**Pull request merged**